demo/quick_start/process_data.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+import sys
+import os
+import operator
+import json
+import random
+from StringIO import StringIO
+import gzip
+import urllib2
+from optparse import OptionParser
+
+defaultFile = 'http://snap.stanford.edu/data/amazon/productGraph/categoryFiles/reviews_Electronics_5.json.gz'
+
+def parse(inputFile, firstNRecords, outputDir, topNWords):
+    """
+    Parse Amazon Reviews data and generate {train,test,pred,dict}.{txt,list}.
+    """
+    train = open(os.path.join(outputDir, 'train.txt'), 'w')
+    test = open(os.path.join(outputDir, 'test.txt'), 'w')
+    pred = open(os.path.join(outputDir, 'pred.txt'), 'w')
+
+    if inputFile.startswith("http"):
+        request = urllib2.Request(inputFile)
+        response = urllib2.urlopen(request)
+        buf = StringIO(response.read())
+        g = gzip.GzipFile(fileobj=buf)
+    else:
+        g = gzip.open(inputFile, 'r')
+
+    written = 0
+    wc = {}
+
+    for l in g:
+        if written >= firstNRecords:
+            break
+
+        o = train
+        u = random.random()
+        if u < 0.01:
+            o = pred
+        elif u < 0.1:
+            o = test
+
+        j = json.loads(l)
+        words = j["reviewText"].lower().split()
+        for w in words:
+            if w not in wc:
+                wc[w] = 1
+            else:
+                wc[w] += 1
+
+        text = " ".join(words)
+        rate = j["overall"]
+        if text:
+            if rate == 5.0:
+                o.write('1\t%s\n' % text)
+                written = written + 1
+            elif rate < 3.0:
+                o.write('0\t%s\n' % text)
+                written = written + 1
```

Comment:
1. We hope in  the rate of positive sample : negative sample is 1:1 in the original process.
2. There are  duplicated samples in reviews_Electronics_5.json.gz.  It's necessary to remove them to make distinct train set and test set.
3.  The moses tools is used to tokenize the words and punctuation. If wedon't want to care about the punctuation, it is ok without moses.
4.  In fact, there is preprocessed data by other people,  http://riejohnson.com/cnn_download.html#sup-paper